### PR TITLE
match custom favicons on the domain

### PR DESCRIPTION
**Pull request by bbraithwaite**
@andrey-p @bsstoner

### CODE REVIEW

File: util.js [hunk 1]
```diff
@@ -236,13 +236,20 @@
...
-            if (isDarkBg && darkVariants.test(customFavicon)) {
-                variant = '.white';
+            // validate domain
+            customFavicon = customFavicon.toString();
+            tokens = sourceUrl.split('.');
+            domainMatch = (tokens.length > 1) ? (tokens[tokens.length - 2] === customFavicon) : true;
+          
+            if (domainMatch) {
+                // check for dark background & available variant
+                if (DDG.settings && DDG.settings.get('kae') === 'd' && darkVariants.test(customFavicon)) {
```
Comment (bbraithwaite) [inline util.js]: The logic to check for dark theme was off. I also moved the check to be inside the IF condition i.e. there's no point checking the value unless it is for a customFavicon that matches.

Comment (andrey-p) [reply util.js]: You're gonna hate this, but this doesn't work on the terminal theme ;)

I think there used to be an `isDark` property exposed on the updater, not sure what happened to it. It's still defined internally, should be pretty easy to re-expose again.

**Review by andrey-p — COMMENTED** [review]
Hopefully minor thing to fix, otherwise :+1:

**Comment by bbraithwaite** [comment]
Updated. I made an internal change for this.

File: util.js [hunk 2]
```diff
@@ -236,13 +236,21 @@
...
-                variant = '.white';
+            // validate domain
+            customFavicon = customFavicon.toString();
+            // if the string matches a favicon domain, remove the TLD (.com) and ensure we match on the domain name only.
+            tokens = sourceUrl.split('.');
+            domainMatch = (tokens.length > 1) ? (tokens[tokens.length - 2] === customFavicon) : true;
+          
+            if (domainMatch) {
+                // check for dark background & available variant
+                if (DDG.settings && DDG.settings && DDG.settings.updater && DDG.settings.updater.isDarkBg && darkVariants.test(customFavicon)) {
```
Comment (andrey-p) [inline util.js]: You've got `DDG.settings` twice there now. I'm just gonna quickly remove one of them before merging.

**Comment by andrey-p** [comment]
:+1:

**Pull request merged**